nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+      #     the API will use different kind of headers to serve audio files
+      #     Allowed values: nginx, apache2
+      #   '';
+      # };
+
+    api = {
+      database_url = mkOption {
+        type = types.str;
+        default = "postgresql://funkwhale@:5432/funkwhale";
+        description = ''
+          Database configuration
+          Examples:
+          postgresql://user:password@host:port/database
+          postgresql://funkwhale:passw0rd@localhost:5432/funkwhale_database
+        '';
+      };
+
+      cache_url = mkOption {
+        type = types.str;
+        default = "redis://127.0.0.1:6379/0";
+        description = ''
+          Cache configuration
+          Examples:
+          redis://host:port/database
+          redis://localhost:6379/0
+        '';
+      };
+
+      media_root = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/media";
+        description = ''
+          Where media files (such as album covers or audio tracks) should be stored on your system?
+          (Ensure this directory actually exists)
+        '';
+      };
+
+      static_root = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/static";
+        description = ''
+          Where static files (such as API css or icons) should be compiled on your system?
+          (Ensure this directory actually exists)
+        '';
+      };
+
+      django_allowed_hosts = mkOption {
+        type = types.str;
+        default = "yourdomain";
+        description = ''
+          Update it to match the domain that will be used to reach your funkwhale instance
+        '';
+      };
+
+      django_secret_key = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Generate one using `openssl rand -base64 45`, for example
+        '';
+      };
+    };
+
+    music_directory_path = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/music";
+        description = ''
+          In-place import settings
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.ntp.enable = true; 
+    services.redis.enable =  true;
+    services.postgresql = {
+      enable = true;
+      # package = pkgs.postgresql100;
+      # enableTCPIP = true;
+      authentication = pkgs.lib.mkOverride 10 ''
+        local all all trust
+        host all all ::1/128 trust
+      '';
+      initialScript = pkgs.writeText "backend-initScript" ''
+        CREATE ROLE funkwhale WITH LOGIN PASSWORD 'funkwhale' CREATEDB;
+        CREATE DATABASE funkwhale WITH ENCODING 'utf8';
+        GRANT ALL PRIVILEGES ON DATABASE funkwhale TO funkwhale;
+      '';
+    };
+    services.nginx = {
+      enable = true;
+      recommendedTlsSettings = true;
+      recommendedOptimisation = true;
+      recommendedGzipSettings = true;
+      recommendedProxySettings = true;
```

Comment:
A service shouldn't set these options on behalf of the user.

Reply:
fixed